internal/locate/region_request.go

```diff
@@ -377,17 +379,21 @@ type tryFollower struct {
 	leaderIdx          AccessIndex
 	lastIdx            AccessIndex
 	labels             []*metapb.StoreLabel
+	// fromOnNotLeader indicates whether the state is changed from onNotLeader.
+	fromOnNotLeader bool
```

Comment:
By adding fromOnNotLeader, it's better to remove `state.fallbackFromLeader` flag

Reply:
And the flag should also set in these places.

https://github.com/tikv/client-go/blob/6f2442951b2bf717081ff383e3164f84675bf3d9/internal/locate/region_request.go#L340

https://github.com/tikv/client-go/blob/6f2442951b2bf717081ff383e3164f84675bf3d9/internal/locate/region_request.go#L355